File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/data/CallTreeListReader.java
package eu.nyerel.panda.ijplugin.data;

import eu.nyerel.panda.monitoringresult.calltree.CallTreeList;

import java.io.*;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public enum CallTreeListReader {

    INSTANCE;

    public CallTreeList read(InputStream is) {
        ObjectInput input;
        try {
            InputStream buffer = new BufferedInputStream(is);
            input = new ObjectInputStream(buffer);
            return ((CallTreeList) input.readObject());
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/data/CallTreeListReader.kt
package eu.nyerel.panda.ijplugin.data

import eu.nyerel.panda.monitoringresult.calltree.CallTreeList

import java.io.*

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
object CallTreeListReader {

    fun read(inputStream: InputStream): CallTreeList {
        val input: ObjectInput
        try {
            val buffer = BufferedInputStream(inputStream)
            input = ObjectInputStream(buffer)
            return input.readObject() as CallTreeList
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }

}
